src/Compilers/CSharp/Portable/Utilities/IValueSet.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// An interface representing a set of values of a specific type.
+    /// </summary>
+    internal interface IValueSet
+    {
+        /// <summary>
+        /// Return the intersection of this value set with another. Both must have been created with the same <see cref="IValueSetFactory{T}"/>.
+        /// </summary>
+        IValueSet Intersect(IValueSet other);
+
+        /// <summary>
+        /// Return this union of this value set with another. Both must have been created with the same <see cref="IValueSetFactory{T}"/>.
+        /// </summary>
+        IValueSet Union(IValueSet other);
+
+        /// <summary>
+        /// Return the complement of this value set.
+        /// </summary>
+        IValueSet Complement();
```

Comment:
>Complement() [](start = 18, length = 12)

Where is this method (or `IValueSet<T>.Complement()`) used?

Reply:
DecisionDagBuilder line 863.

---
In reply to: [378543186](https://github.com/dotnet/roslyn/pull/41503#discussion_r378543186) [](ancestors = 378543186)